src/mbed_tools/devices/_internal/resolve_board.py

```diff
@@ -61,12 +57,12 @@ def resolve_board(candidate: CandidateDevice) -> Board:
 
```

Comment:
If we get UnknownBoard from the provided online_id, it looks like we'd raise an exception here and not fall back to using the USB serial number. Is that correct?